src/database/sqlx-mysql/src/mysql_plugin.rs

```diff
@@ -0,0 +1,92 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use database_common::{
+    DatabaseConfiguration,
+    DatabaseError,
+    DatabaseTransactionManager,
+    TransactionSubject,
+};
+use dill::*;
+use kamu_core::{InternalError, ResultIntoInternal};
+use sqlx::MySqlPool;
+
+use crate::{MySqlAccountRepository, MySqlConnectionPool, MySqlTransaction};
+
+/////////////////////////////////////////////////////////////////////////////////////////
+
+pub struct MySqlPlugin {}
+
+#[component(pub)]
+#[interface(dyn DatabaseTransactionManager)]
+impl MySqlPlugin {
+    pub fn new() -> Self {
+        Self {}
+    }
+
+    pub fn init_database_components(
+        catalog_builder: &mut CatalogBuilder,
+        db_configuration: &DatabaseConfiguration,
+    ) -> Result<(), DatabaseError> {
+        let mysql_pool = Self::open_mysql_pool(db_configuration)?;
+
+        catalog_builder.add::<Self>();
+        catalog_builder.add_builder(MySqlConnectionPool::builder().with_mysql_pool(mysql_pool));
+        catalog_builder.add::<MySqlAccountRepository>();
```

Comment:
It looks wrong that a very generic transaction manager component knows something about "accounts"

Reply:
I do not have any other place yet, what's so surprising.

Reply:
I think this logic should simply reside in `app.rs` along with the rest of catalog configuration.

Reply:
Fixed